widget/entry-with-completion.go

```diff
@@ -0,0 +1,141 @@
+package widget
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/widget"
+)
+
+// CompletionEntry is an Entry with options displayed in a PopUpMenu.
+//
+// Example:
+//
+//    entry := NewCompletionEntry()
+//    entry.OnChanged = func(s string) {
+//        if len(s) < 3 {
+//            entry.HideCompletion()
+//        }
+//        entry.SetItems([]string{"foo", "bar", "baz"}
+//        entry.ShowCompletion()
+//    }
+type CompletionEntry struct {
+	*widget.Entry
+	menu  *fyne.Menu
+	cmenu *widget.PopUpMenu
+	items []string
+	pause bool
+}
+
+// NewCompletionEntry creates a new CompletionEntry which creates a popup menu that responds to keystrokes to navigate through the items without losing the editing ability of the text input..
+func NewCompletionEntry(options []string) *CompletionEntry {
+	c := &CompletionEntry{}
+	// c.ExtendBaseWidget(c)
+	c.Entry = widget.NewEntry()
+	c.items = options
+	c.pause = false
+	return c
+}
+
+// ShowCompletion displays the completion menu
+func (c *CompletionEntry) ShowCompletion() {
+	if c.pause {
+		return
+	}
+	items := make([]*fyne.MenuItem, 0)
```

Comment:
Why make the slice to zero length? I think it would be better to make it to the length of items. Then you can workaround having to append each time. Should be a bit faster.

Reply:
I mimic what it's done in the EntrySelect from Fyne. I agree that I can use the size of items to prepare the capacity.

Reply:
What I meant was that there is no point in using `make()` to create an empty slice. It is better to just use `var items []*fyne.MenuItem` in that case. SelectEntry actually does prepare the capacity as of https://github.com/fyne-io/fyne/pull/2169. Thanks for updating it to use that.